bcycle-backend/Services/TripService.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using bcycle_backend.Data;
+using bcycle_backend.Models;
+using bcycle_backend.Models.Dto;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+
+namespace bcycle_backend.Services
+{
+    public class TripService
+    {
+        private readonly DbSet<Trip> _tripsDbSet;
+        private readonly IConfiguration _configuration;
+        private readonly BCycleContext _dbContext;
+
+        public TripService(IConfiguration configuration, BCycleContext context)
+        {
+            _configuration = configuration;
+            _tripsDbSet = context.Trips;
+            _dbContext = context;
+        }
+
+        public IQueryable<TripDto> GetAll(string userId) =>
+            _tripsDbSet
+                .Where(t => t.UserId == userId)
+                .Include(t => t.TripPhotos)
+                .Include(t => t.TripPoints)
+                .OrderByDescending(t => t.Started)
+                .Select(t => t.AsDto());
+
+        public Task<Trip> GetUserTripAsync(int tripId, string userId) =>
+            _tripsDbSet
+                .Where(t => t.Id == tripId && t.UserId == userId)
+                .Include(t => t.TripPhotos)
+                .Include(t => t.TripPoints)
+                .FirstOrDefaultAsync();
+
+
+        public async Task<Trip> SaveTripAsync(TripDto tripDto, string userId)
+        {
+            var trip = tripDto.AsTrip();
+            trip.UserId = userId;
+
+            _tripsDbSet.Add(trip);
+            await _dbContext.SaveChangesAsync();
+            return trip;
+        }
+
+        public async Task<Trip> RemoveAsync(int tripId, string userId)
+        {
+            var trip = await GetUserTripAsync(tripId, userId);
+            return trip == null ? null : await Remove(trip);
+
+            async Task<Trip> Remove(Trip tripToDelete)
```

Comment:
Do we really need a local function here? Can’t we just guard with `if (trip == null) return null;`, as we do in `PutPhotoAsync`?

Reply:
We can, but we also can do it like this. I wanted to get rid of if statement, since they make the code uglier.

Reply:
I think that an `if` statement is much prettier and more obvious than introducing a local function called in one condition. Code style is subjective. My opinion is that readable code is better than clever code. Plus some people claim less indentation = better.

Reply:
I'mo that's readable code. Also it looks more concise. There's nothing clever in using language feature in a simple way.

Reply:
Also there's same amount of indentation in both cases

Reply:
There’s less indented lines:

```csharp
        public async Task<Trip> RemoveAsync(int tripId, string userId)
        {
            var trip = await GetUserTripAsync(tripId, userId);
            if (trip == null)
            {
                return null;
            }
            _tripsDbSet.Remove(tripToDelete);
            await _dbContext.SaveChangesAsync();
            return tripToDelete;
        }
```

This code has the same number of lines and is immediately obvious.

Reply:
Indentation level is the same thats what i meant. Cmon theres no point in arguing over such irrelevant thing :)

Reply:
If we gonna argue about every style related case, where one approach isn't objectively better than the other, we won't finish this project too soon